onnxruntime/contrib_ops/cuda/bert/rotary_embedding_impl.cu

```diff
@@ -82,20 +92,30 @@ Status LaunchRotaryEmbeddingKernel(
     const int sequence_length,
     const int num_heads,
     const int head_size,
+    const int rotary_embedding_dim,
     const int max_sequence_length,
     const int position_ids_format,
     const bool interleaved,
     const int max_threads_per_block,
     const bool transposed) {
-
-  constexpr int smem_size = 0;
-  const dim3 grid(num_heads, sequence_length, batch_size);
-  const dim3 block(head_size, 1, 1);
-
   // Note: Current implementation assumes head_size <= max_threads_per_block
   // because head_size is currently large for LLaMA-2. For smaller head_size
   // and num_heads values, we can create a block as `block(num_heads, head_size, 1)`
   // instead. This will require kernel changes to support.
+  ORT_ENFORCE(head_size <= max_threads_per_block,
+              "Rotary embedding dim must be <= max_threads_per_block");
+
+  int tpb = head_size;
```

Comment:
What is this logic for?

Reply:
it is to ensure the block_size be multiple of warp size.

Reply:
why not just (head_size + 31)/32*32 work?

Reply:
it uses fewer instructions but both work

Reply:
changed to (head_size + 31)/32*32

Reply:
it doesn't actually use less instructions. (head_size + 31)/32*32 uses 1 add, 1 right shift and 1 left shift, in total, 3 instructions.